### PR TITLE
Fix test to use TokenRequest API for garden clusters >=1.24

### LINKED ISSUE #6975: Test machinery tests are failing after upgrading virtual garden api-server to 1.24.7
**How to categorize this issue?**

/area testing
/kind api-change

**What happened**:
With 1.24 the service account secrets are no longer created automatically. As result, the test machinery tests are failing

**What you expected to happen**:
Testmachinary code to be adapted https://github.com/gardener/gardener/blob/fc2fa73ffd200061a22e3ed8d052b483c940dd35/test/testmachinery/gardener/security/rbac.go#L87-L98

**How to reproduce it (as minimally and precisely as possible)**:
Check Failed builds on Dev. For example concourse build number: 159

**Anything else we need to know?**:

**Environment**:

- Gardener version: v1.59.1
- Kubernetes version: 1.24.7

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area testing
/kind bug

**What this PR does / why we need it**:
Fix testmachinery test to always use TokenRequest API and to properly clean up created service account

**Which issue(s) this PR fixes**:
Fixes #6975 

**Special notes for your reviewer**:

**Release note**:

```bugfix operator
Fix garden RBAC test to use `TokenRequest` API.
```

### CODE REVIEW

**Comment by dguendisch** [comment]
/lgtm
thank you!

**Review by rfranzke — COMMENTED** [review]
/approve

File: test/testmachinery/gardener/security/rbac.go [hunk 1]
```diff
@@ -84,20 +86,36 @@ var _ = ginkgo.Describe("RBAC testing", func() {
 			framework.ExpectNoError(f.GardenClient.Client().Delete(ctx, serviceAccount))
 		}()
 
+		gardenVersion := f.GardenClient.Version()
+		gardenLess124, err := version.CheckVersionMeetsConstraint(gardenVersion, "< 1.24.0")
+		framework.ExpectNoError(err)
+
```
Comment (timebertt) [inline test/testmachinery/gardener/security/rbac.go]: Can you simplify this to always use the token request API?
The minimum supported garden version is 1.20 which should offer the token request API by default.

**Review by timebertt — CHANGES_REQUESTED** [review]

**Comment by vpnachev** [comment]
/lgtm cancel 

The test is leaking ServiceAccounts, I will try to fix this, too.

**Comment by timebertt** [comment]
There is also https://github.com/gardener/gardener/blob/b7c070b02657b224d55ed73a54207d21ae3d976f/test/utils/shoots/access/serviceaccount.go including helpers for cleanup. You might want to reuse/share some code.

**Comment by vpnachev** [comment]
/title Fix testmachinery test to always use TokenRequest API and to properly clean up created service account

File: test/testmachinery/gardener/security/rbac.go (hunk 1 above)
Comment (vpnachev) [reply test/testmachinery/gardener/security/rbac.go]: Done.

**Comment by vpnachev** [comment]
I am curious, can I lgtm this myself?
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**